Ginger/GingerCore/Drivers/SeleniumDriver.cs

```diff
@@ -6301,23 +6302,50 @@ bool IWindowExplorer.IsElementObjectValid(object obj)
             return true;
         }
 
-        public void TestElementLocators(ObservableList<ElementLocator> elementLocators)
+        bool IWindowExplorer.TestElementLocators(ObservableList<ElementLocator> elementLocators, bool GetOutAfterFoundElement = false)
         {
-            foreach (ElementLocator el in elementLocators)
-                el.LocateStatus = ElementLocator.eLocateStatus.Pending;
+            try
+            {
+
+                foreach (ElementLocator el in elementLocators)
+                    el.LocateStatus = ElementLocator.eLocateStatus.Pending;
 
-            foreach (ElementLocator el in elementLocators.Where(x=>x.Active==true).ToList())
-            {                
-                if (LocateElementByLocator(el, true) != null)
+                List<ElementLocator> activesElementLocators = elementLocators.Where(x => x.Active == true).ToList();
+                Driver.Manage().Timeouts().ImplicitWait = new TimeSpan(0, 0, 0);
+                foreach (ElementLocator el in activesElementLocators)
                 {
-                    el.LocateStatusError = string.Empty;
-                    el.LocateStatus = ElementLocator.eLocateStatus.Passed;                   
+                    if (LocateElementByLocator(el, true) != null)
+                    {
+                        el.StatusError = string.Empty;
+                        el.LocateStatus = ElementLocator.eLocateStatus.Passed;
+                        if (GetOutAfterFoundElement)
+                        {
+                            return true;
+                        }
+                    }
+                    else
+                    {
+                        el.LocateStatus = ElementLocator.eLocateStatus.Failed;
+                    }
+                }
+
+                Driver.Manage().Timeouts().ImplicitWait = (TimeSpan.FromSeconds((int)ImplicitWait));
+
+                if (activesElementLocators.Where(x => x.LocateStatus == ElementLocator.eLocateStatus.Passed).Count() > 0)
+                {
+                    return true;
                 }
                 else
                 {
-                    el.LocateStatus = ElementLocator.eLocateStatus.Failed;
+                    return false;
                 }
             }
+            catch (Exception ex)
+            {
+                Driver.Manage().Timeouts().ImplicitWait = (TimeSpan.FromSeconds((int)ImplicitWait));
+                throw ex;
+            }
+            
```

Comment:
Add Finally block and place the Driver.Manage().Timeouts().ImplicitWait to defualt in it, so in any case it will be set back, this is the correct way of doing it instead excpetion catch and then throw, you can do only try{}Finall{} without catch